### PR TITLE
Filtering inconsistent 2-view relative poses after rotation averaging

### PR DESCRIPTION
**Pull request by ayushbaid**
Adding a filtering step where we only keep the 2-view geometries which are consistent with the global rotations computed by Shonan

### CODE REVIEW

**Comment by ayushbaid** [comment]
This PR is not ready for review. I have opened it to run benchmarks.

**Comment by johnwlambert** [comment]
Thanks for adding this Ayush. I see on skydio 32 -- 
```
[2021-09-07 09:13:36,947 INFO multi_view_optimizer.py line 134 6770] Dropped 312/375 unit-translations post rotation averaging
```
my guess is we need to increase the threshold, and we should look at the relative error. vs. GT error scatter plot

File: gtsfm/multi_view_optimizer.py [hunk 1]
```diff
@@ -6,16 +6,22 @@
...
+import gtsfm.utils.geometry_comparisons as geom_utils
+import gtsfm.utils.logger as logger_utils
 import gtsfm.utils.metrics as metrics_utils
 from gtsfm.averaging.rotation.rotation_averaging_base import RotationAveragingBase
 from gtsfm.averaging.translation.translation_averaging_base import TranslationAveragingBase
 from gtsfm.bundle.bundle_adjustment import BundleAdjustmentOptimizer
 from gtsfm.data_association.data_assoc import DataAssociation
 from gtsfm.evaluation.metrics import GtsfmMetricsGroup
 
+POST_ROTATION_AVEGARING_OUTLIER_REMOVAL_ANGULAR_THRESHOLD_DEGREES = 10
```
Comment (johnwlambert) [inline gtsfm/multi_view_optimizer.py:21]: small typo up here on "Averaging"

**Comment by ayushbaid** [comment]
# Number of cameras dropped across benchmarks

| Benchmark | Number of unit translations dropped | 
| -------------- | --------- |
| Skydio-32 deep front end | 117/375 |
| Door-12 sift front end | 0/66 |


# Metrics for skydio-32

## Baseline



Metric name | Value
-- | --
num_rotations_computed | 32
mean_rotation_error_angle_deg | 7.95789
num_total_1dsfm_measurements | 375
num_inlier_1dsfm_measurements | 252
num_outlier_1dsfm_measurements | 123
1dsfm_precision_5_deg | 0.670635
1dsfm_recall_5_deg | 0.725322
num_translations_estimated | 32
mean_1dsfm_inlier_angular_errors_deg | 10.7212
mean_1dsfm_outlier_angular_errors_deg | 50.8471
mean_translation_angle_error_deg | 11.0073
mean_translation_error_distance | 0.763393



## New

 Metric name | Value 
 -- | -- 
num_rotations_computed | 32
mean_rotation_error_angle_deg | 7.96289
num_total_1dsfm_measurements | 258
num_inlier_1dsfm_measurements | 173
num_outlier_1dsfm_measurements | 85
1dsfm_precision_5_deg | 0.803468
1dsfm_recall_5_deg | 0.755435
num_translations_estimated | 32
mean_1dsfm_inlier_angular_errors_deg | 12.17
mean_1dsfm_outlier_angular_errors_deg | 45.9713
mean_translation_angle_error_deg | 156.777
mean_translation_error_distance | 3.59323


@akshay-krishnan I think this does not help us significantly. Should we close it?

**Comment by ayushbaid** [comment]
Closing as no significant improvements

**Pull request closed without merging**